src/main/java/algorithms/curated170/medium/AddBoldTaginString.java

```diff
@@ -0,0 +1,38 @@
+package algorithms.curated170.medium;
+
+public class AddBoldTaginString {
+
+    public String addBoldTag(String s, String[] dict) {
+        StringBuilder sb = new StringBuilder();
+        boolean arr[] = new boolean[s.length()];
+        for(String d : dict){
+            int start = s.indexOf(d);
+            while(start != -1){
+                int length = start+ d.length();
+                for(int i = start; i < length; i++){
+                    arr[i] = true;
+                }
+                start = s.indexOf(d, start + 1);
```

Comment:
Is this used to check the recurrences of the same string `d` inside of the string `s` after its previously detected index?